src/test/ui/consts/const-eval/issue-62272.rs

```diff
@@ -1,9 +1,11 @@
-// run-pass
+// `loop`s unconditionally-broken-from used to be allowed in constants, but are now forbidden by
+// the HIR const-checker.
+//
+// See https://github.com/rust-lang/rust/pull/66170 and
+// https://github.com/rust-lang/rust/issues/62272.
 
-// Tests that `loop`s unconditionally-broken-from are allowed in constants.
-
-const FOO: () = loop { break; };
+const FOO: () = loop { break; }; //~ ERROR `loop` is not allowed in a `const`
 
 fn main() {
-    [FOO; { let x; loop { x = 5; break; } x }];
+    [FOO; { let x; loop { x = 5; break; } x }]; //~ ERROR `loop` is not allowed in a `const`
```

Comment:
How do we feel about the term `const` for an array initializer? Is there a preexisting convention here? It's pretty easy to change.

Reply:
Maybe @estebank has thoughts?

Reply:
I would prefer us to be _correct_ mentioning the array length is a const context, but this should be ok for now, we have plenty of other places where we are inconsistent about calling this out.

Reply:
Oh, also, I think const generic arguments are also `anon_const`, so this isn't that easy. Probably the right way is to add an `AnonConstKind` to `AnonConst` in `hir/mod.rs`. With that in mind, I'm inclined to fix this later :).